Allison Forte

September 16, 2022

DSC 630

# 3.2 Assignment: Using Data to Improve MLB Attendance

Using data on the Los Angeles Dodgers Major League Baseball (MLB) team, make a recommendation to management on how to improve attendance. 
Tell a story with your analysis and clearly explain the steps you take to arrive at your conclusion. 
This is an open-ended question, and there is no one right answer. 
You are welcome to do additional research and/or use domain knowledge to assist your analysis, but clearly state any assumptions you make.

You can use R or Python to complete this assignment. Submit your code and output to the submission link. Make sure to add comments to all of your code and to document your steps, process, and analysis.

In [2]:
# Load data

import pandas as pd
import numpy as np

df = pd.read_csv('/Users/allison.forte/Downloads/dodgers-2022.csv')
print(df.head())
df.describe()

  month  day  attend day_of_week opponent  temp   skies day_night cap shirt  \
0   APR   10   56000     Tuesday  Pirates    67  Clear        Day  NO    NO   
1   APR   11   29729   Wednesday  Pirates    58  Cloudy     Night  NO    NO   
2   APR   12   28328    Thursday  Pirates    57  Cloudy     Night  NO    NO   
3   APR   13   31601      Friday   Padres    54  Cloudy     Night  NO    NO   
4   APR   14   46549    Saturday   Padres    57  Cloudy     Night  NO    NO   

  fireworks bobblehead  
0        NO         NO  
1        NO         NO  
2        NO         NO  
3       YES         NO  
4        NO         NO  


day        attend       temp
count  81.000000     81.000000  81.000000
mean   16.135802  41040.074074  73.148148
std     9.605666   8297.539460   8.317318
min     1.000000  24312.000000  54.000000
25%     8.000000  34493.000000  67.000000
50%    15.000000  40284.000000  73.000000
75%    25.000000  46588.000000  79.000000
max    31.000000  56000.000000  95.000000

## Findings and analysis

Baseball schedules depend on many factors and require games against all teams. The opponent, while it can affect the attendance, is not something that can be easily changed. Factors including time of day and day of week can more easily be changed but are still often determined by more factors than just management decision. The easiest factors for management to change are whether games have free give aways or additional perks. This dataset has information about cap, shirt, and bobblehead giveaways as well as fireworks.

The max attendance in this dataset was 56,000 people and the minimum attendance was 24,312 people with all the data provided. The mean attendance was 41040 people. Tuesdays are the most attended day of the week with an average attendance of 47741.

In [4]:
# calculate the average attendance for day games vs night games

table1 = pd.pivot_table(df, values=['attend'], index=['day_night'],
                    aggfunc={'attend': np.mean})

table1

attend
day_night              
Day        41793.266667
Night      40868.893939

With the current schedule and attendance, we can see that the time of day alone does not make a significant difference in the overall attendance of the game. Adjusting the time of games will likely not improve attendance in a significant way. 

In [7]:
# calculate attendance by day of week

table1 = pd.pivot_table(df, values=['attend'], index=['day_of_week'],
                    aggfunc={'attend': np.mean}, sort = False)

table1

attend
day_of_week              
Tuesday      47741.230769
Wednesday    37585.166667
Thursday     40407.400000
Friday       40116.923077
Saturday     43072.923077
Sunday       42268.846154
Monday       34965.666667

Additionally, looking the attendance by day of the week, it is clear that Monday games have the lowest attendance with Wednesday games being a close second. To improve attendance, fewer games should be scheduled on Mondays and Wednesdays with a preferred game day of Tuesday followed by Saturday then Sunday. With this information, it is important to remember that adjusting the day of a game is not as simple as a quick switch. The opponent's schedule as well as other factors must be considered before this recomendation can be enacted. 

Looking at thefactors that are more easily controlled, caps, shirts, fireworks, and bobbleheads all have different impacts on attendance. 

In [11]:
# Check the overall impact of caps

cap = pd.pivot_table(df, values=['attend'], index=['cap'],
                    aggfunc={'attend': np.mean})                 
cap

attend
cap              
NO   41112.240506
YES  38189.500000

In [12]:
# Check the impact of caps by day of the week

cap2 = pd.pivot_table(df, values=['attend'], index=['cap', 'day_of_week'],
                    aggfunc={'attend': np.mean})
cap2

attend
cap day_of_week              
NO  Friday       40116.923077
    Monday       34965.666667
    Saturday     43072.923077
    Sunday       42250.000000
    Thursday     40407.400000
    Tuesday      48896.000000
    Wednesday    37585.166667
YES Sunday       42495.000000
    Tuesday      33884.000000

Based on the data provided by this dataset, caps should no longer be given out. Overall caps bring down attendance. Caps on Tuesdays significantly reduce attendance. Caps on Sunday lead to a very small increase in attendance. The expense spent on caps should be diverted to a more productive incentive.

In [14]:
# Check the overall impact of shirts

shirt = pd.pivot_table(df, values=['attend'], index=['shirt'],
                    aggfunc={'attend': np.mean})
shirt

attend
shirt              
NO     40824.551282
YES    46643.666667

In [15]:
# Check the impact of shirts by day of the week
                    
shirt2 = pd.pivot_table(df, values=['attend'], index=['day_of_week','shirt'],
                    aggfunc={'attend': np.mean})
shirt2

attend
day_of_week shirt              
Friday      NO     40116.923077
Monday      NO     33548.090909
            YES    50559.000000
Saturday    NO     43072.923077
Sunday      NO     41728.500000
            YES    48753.000000
Thursday    NO     40407.400000
Tuesday     NO     48334.750000
            YES    40619.000000
Wednesday   NO     37585.166667

Shirts, unlike caps, have an overall positive impact on attendance with an average attendance on shirt nights being 46643 vs 40824 without shirts. Shirts are a significant draw on Sunday and Monday but have an opposite affect on Tuesdays. Shirts, like caps, should not be given out on Tuesdays. 

In [17]:
# Check the overall impact of fireworks

fireworks = pd.pivot_table(df, values=['attend'], index=['fireworks'],
                    aggfunc={'attend': np.mean})
fireworks

attend
fireworks              
NO         41032.179104
YES        41077.857143

In [18]:
# Check the impact of fireworks by day of the week
                
fireworks2 = pd.pivot_table(df, values=['attend'], index=['day_of_week','fireworks'],
                    aggfunc={'attend': np.mean})
fireworks2

attend
day_of_week fireworks              
Friday      YES        40116.923077
Monday      NO         34965.666667
Saturday    NO         43072.923077
Sunday      NO         42268.846154
Thursday    NO         40407.400000
Tuesday     NO         47741.230769
Wednesday   NO         36132.000000
            YES        53570.000000

Overall, fireworks do not have a significant impact on attendance bringing the average attendance from 41032 to 41078. However, the fireworks on Wendesdays bring significantly more people increasing the average Wednesday night attendance from 36132 to 53570. There is not enough data to conclude whether frieworks on Fridays increase attendance. Based on this information, I would recomend moving half of the Friday fireworks shows to Wednesday or Monday which would have 2 positive effects. First, it will likely increase attendance on those days and second, it would allow the impact of fireworks on Friday to be further studied. 

In [23]:
# Check the overall impact of bobbleheads

bobblehead = pd.pivot_table(df, values=['attend'], index=['bobblehead'],
                    aggfunc={'attend': np.mean})
bobblehead

attend
bobblehead              
NO          39137.928571
YES         53144.636364

In [24]:
# Check the impact of bobbleheads by day of the week

bobblehead2 = pd.pivot_table(df, values=['attend'], index=['day_of_week','bobblehead'],
                    aggfunc={'attend': np.mean})
bobblehead2

attend
day_of_week bobblehead              
Friday      NO          40116.923077
Monday      NO          34965.666667
Saturday    NO          41062.909091
            YES         54128.000000
Sunday      NO          41178.000000
            YES         55359.000000
Thursday    NO          32803.333333
            YES         51813.500000
Tuesday     NO          43326.714286
            YES         52891.500000
Wednesday   NO          37585.166667

Bobbleheads have the most significant impact on attendance overall. The average with bobblehead giveaways increases from 39138 to 53145. Given Monday has the lowest attendance of the whole week, adding some bobblehead days on Mondays will likely show a positive impact. Wednesdays would also likely benefit from bobbleheads.

Overall, based on what has been seen with this dataset, I would make several recomendations for improving attendance. First, whenever possible, avoid scheduling Monday and Wednesday games. In addition, ensure there is a game every Tuesday to maximize the most attended day of the week.

Secondly, think more strategically about the money spent on incentives. Do not offer any incentives on Tuesday. For other days of the week, avoid spending money on caps and try instead to introduce some shirt days on Thursdays. 

Fireworks should be increased on Wednesdays. Moving some Friday fireworks days to Wednesday would be a good option to likely increase attendance on Wednesday and see whether they are necessary on Fridays. 

Bobbleheads are a big draw for fans. Using funds from caps and avoided Tuesday incentives to add more bobblehead days will likely be a better return on investment. 

Without knowing the cost of each incentive, it is hard to know the exactly what the impact of specific changes will be. Additional information about the cost of each incentive and the overall budget for seasonal incentives would help inform a concrete plan for next season. 